### PR TITLE
[StableX] adding docker-compose config

### PR DESCRIPTION
**Pull request by fleupold**
This PR adds the docker compose service for stableX. I moved the common rust-service configuration into another file for it to be shared between dfusion stablex and the graph-listener

### Test Plan

`docker-compose up stablex`

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,6 +1,18 @@
 version: '3.6'
-services:
+x-rust-service:
+  &rust-service
```
Comment (bh2smith) [inline docker-compose.yml]: Is this some sort of tag? referencing the `*rust-service`s appearing later?

**Comment by bh2smith** [comment]
After building and running the above command, I first got 

```
stablex_1         | 2019-09-25 12:09:01,504 ERROR [stablex] StableXDriver error: Driver Error: [please ensure the contract and method you're calling exist! failed to decode empty bytes. if you're using jsonrpc this is likely due to jsonrpc returning `0x` in case contract or method don't exist]
stablex_1         | 2019-09-25 12:09:06,520 ERROR [stablex] StableXDriver error: Driver Error: [Transport error: Error(Io, Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" })]
```

Then I migrated the contracts and got the same as in the other PR #255.

```
stablex_1         | 2019-09-25 12:10:06,818 ERROR [stablex] StableXDriver error: Driver Error: [RPC error: Error { code: ServerError(-32000), message: "VM Exception while processing transaction: revert Trying to get first from empty set", data: Some(Object({"0x70b898028f3c27575c44f818938917d7785c54a52e39dc70f6fb438c56ba5a9b": Object({"error": String("revert"), "program_counter": Number(4512), "return": String("0x08c379a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000022547279696e6720746f206765742066697273742066726f6d20656d70747920736574000000000000000000000000000000000000000000000000000000000000"), "reason": String("Trying to get first from empty set")}), "stack": String("o: VM Exception while processing transaction: revert Trying to get first from empty set\n    at Function.o.fromResults (/app/ganache-core.docker.cli.js:10:82491)\n    at /app/ganache-core.docker.cli.js:25:152176\n    at /app/ganache-core.docker.cli.js:37:1958262\n    at /app/ganache-core.docker.cli.js:2:80687\n    at /app/ganache-core.docker.cli.js:2:66123\n    at f (/app/ganache-core.docker.cli.js:2:69298)\n    at s (/app/ganache-core.docker.cli.js:2:69216)\n    at /app/ganache-core.docker.cli.js:2:69010\n    at /app/ganache-core.docker.cli.js:2:80667\n    at /app/ganache-core.docker.cli.js:37:1958207\n    at /app/ganache-core.docker.cli.js:37:1319544\n    at /app/ganache-core.docker.cli.js:2:66123\n    at f (/app/ganache-core.docker.cli.js:2:69298)\n    at /app/ganache-core.docker.cli.js:2:69340\n    at Ht (/app/ganache-core.docker.cli.js:2:78701)\n    at Object.<anonymous> (/app/ganache-core.docker.cli.js:2:69429)\n    at /app/ganache-core.docker.cli.js:37:1319465\n    at /app/ganache-core.docker.cli.js:2:69010\n    at i (/app/ganache-core.docker.cli.js:2:84065)\n    at /app/ganache-core.docker.cli.js:37:1319446\n    at i (/app/ganache-core.docker.cli.js:2:84037)\n    at /app/ganache-core.docker.cli.js:37:1319417"), "name": String("o")})) }]
```

**Comment by josojo** [comment]
I am stuck here as well but at another point

```
dex-services git:(stablex_docker) ✗ docker-compose up stablex   
dex-services_ganache-cli_1 is up-to-date
Starting dex-services_stablex_1 ... done
Attaching to dex-services_stablex_1
stablex_1         |     Finished dev [unoptimized + debuginfo] target(s) in 1.23s
stablex_1         |      Running `target/debug/stablex`
stablex_1         | 2019-09-25 12:49:48,250 INFO  [driver::util] Using naive price finder
stablex_1         | 2019-09-25 12:49:48,263 ERROR [stablex] StableXDriver error: Driver Error: [Invalid name `getCurrentBatchId`]
stablex_1         | 2019-09-25 12:49:53,264 ERROR [stablex] StableXDriver error: Driver Error: [Invalid name `getCurrentBatchId`]
stablex_1         | 2019-09-25 12:49:58,266 ERROR [stablex] StableXDriver error: Driver Error: [Invalid name `getCurrentBatchId`]
```
I get this error messages although 
- getCurrentBatchId is in my smart contracts
- the address during migration is the same as provided in the env variable

though i could not checkout the correct dex-contract branch:
```
git checkout fc4a6d16c36763e9fde09972add876f00478ba68
fatal: reference is not a tree: fc4a6d16c36763e9fde09972add876f00478ba68

or 

git pull origin fc4a6d16c36763e9fde09972add876f00478ba68
error: Server does not allow request for unadvertised object fc4a6d16c36763e9fde09972add876f00478ba68

```

File: docker-compose.yml (hunk 1 above)
Comment (fleupold) [reply docker-compose.yml]: Check out https://docs.docker.com/compose/compose-file/#extension-fields

**Comment by bh2smith** [comment]
@josojo I managed to get the contracts right by checking out master and pulling whatever was new. This was somehow aligned with the expected contracts even though I was also struggling to find the right commit hash.

**Pull request merged**